src/main/java/com/zendesk/maxwell/RowMap.java

```diff
@@ -234,16 +238,40 @@ public Object getData(String key) {
 		return this.data.get(key);
 	}
 
+
+	public long getApproximateSize() {
+		return approximateSize;
+	}
+
+	private long approximateKVSize(String key, Object value) {
+		this.approximateSize += 40; // overhead.  Whynot.
+		this.approximateSize += key.getBytes().length;
+
+		long length;
+
+		if ( value instanceof String ) {
+			length = ((String) value).getBytes().length * 2;
+		} else {
+			length = 64;
+		}
+
+		return length;
+	}
+
 	public void putData(String key, Object value) {
-		this.data.put(key,  value);
+		this.data.put(key, value);
+
+		this.approximateSize += approximateKVSize(key, value);
```

Comment:
`approximateKVSize` updates `approximateSize ` and here it also updates `approximateSize` with the result `approximateKVSize`; are we double updating it?